packages/orchestration/src/service.js

```diff
@@ -51,37 +53,62 @@ const getPower = (powers, name) => {
 };
 
 export const OrchestrationI = M.interface('Orchestration', {
-  makeAccount: M.callWhen(M.string(), M.string()).returns(
-    M.remotable('ChainAccount'),
-  ),
-  provideICQConnection: M.callWhen(M.string()).returns(
-    M.remotable('Connection'),
+  makeAccount: M.call(M.string(), M.string()).returns(M.promise()),
```

Comment:
`makeAccount` is no longer an async function, so I didn't think `.callWhen()` was appropriate. Reverted in [1471410](https://github.com/Agoric/agoric-sdk/pull/9454/commits/1471410905342adfa67f9f6b52e704821630307e)

Reply:
I believe that `.callWhen` is not simple validation and indeed make the function become async. @erights for confirmation, but we may need to manually validate this, or find a way to make this pattern check compatible with upgradable methods.

Reply:
Isn't the problem whether an `await` happens in the method? If the JS code that this is guarding doesn't have the `async` keyword then it can't await.

How could the guard change that?

Reply:
The problem is doing any promise "then" if the promise may not be resolved until a future crank. Whether that's done by an `await`, by a `when` or any other mechanism that under the hood wraps `promise.then()`. `callWhen` is such a mechanism as, from what I understand, it internally `when` the arguments and delays the invocation of the user function until after the arguments have settled.